models/edr/dbt_artifacts/dbt_artifacts_state.sql

```diff
@@ -0,0 +1,22 @@
+{{
+  config(
+    materialized = 'view',
+    bind=False
+  )
+}}
+
+{% set hash_agg_expr %}
+md5({{ elementary.listagg("artifact_hash") }})
```

Comment:
From Snowflake's docs:
> If you do not specify the WITHIN GROUP (<orderby_clause>), the order of elements within each list is unpredictable. (An ORDER BY clause outside the WITHIN GROUP clause applies to the order of the output rows, not to the order of the list elements within a row.)

From BigQuery's docs:
> The ORDER BY clause is supported only for aggregate functions that depend on the order of their input. For those functions, if the ORDER BY clause is omitted, the output is nondeterministic.

It seems that not using `ORDER BY` results in nondeterministic order.
However, in practice, doing the `ORDER BY` on the source table seems to work.

The reason for not doing it within the `listagg()` is that Databricks doesn't support it.